hax/minitrees.py

```diff
@@ -161,16 +218,25 @@ def get(run_name, treemaker, force_reload=False):
     if not hasattr(treemaker, '__version__'):
         raise AttributeError("Please add a __version__ attribute to treemaker %s." % treemaker_name)
     minitree_filename = "%s_%s.root" % (run_name, treemaker_name)
+    if save_pickle:
+        minitree_pickle_filename = "%s_%s.pkl" % (run_name, treemaker_name)
 
     # Do we already have this minitree? And is it good?
     minitree_path = _check_minitree_path(minitree_filename, treemaker, run_name,
                                          force_reload=force_reload)
     if minitree_path is not None:
-        return minitree_path
+        empty_frame = pd.DataFrame()
+        return minitree_path, empty_frame
 
     # We have to make the minitree file
     # This will raise FileNotFoundError if the root file is not found
     skimmed_data = treemaker().get_data(run_name)
+    
+    # Setting save_arrays to True if any arrays/vectors in DataFrame
+    for branch_name in list(skimmed_data):
+        if hasattr(skimmed_data[branch_name][0], "__len__"):
```

Comment:
Strings and bytes also have the __len__ attribute, but you probably don't want them to trigger your array saving logic. You could broaden the check to
```
x = skimmed_data[branch_name][0]
if hasattr(x, "__len__") and not isinstance(x, (str, bytes)):
    ...
```
This will still raise a rather scary/uninformative IndexError if there is no data in skimmed_data -- but that's a very rare edge case. There's a another instance of this check in dataframe_to_root (might be worth factoring out to a function).